### PR TITLE
DM-15431: expand pybind11 interface

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

**Comment by PaulPrice** [comment]
The commit messages violate [the guidelines for commit messages](https://developer.lsst.io/work/flow.html#git-commit-message-best-practices):
* the first is not in the imperative tense;
* the second is vague and seems to refer to some other work ("more") that requires context to understand;
* neither contain any information that informs the reader about what has been changed, and why it's necessary.

File: python/lsst/jointcal/gtransfo.cc [hunk 1]
```diff
@@ -49,6 +49,11 @@ void declareGtransfo(py::module &mod) {
             "inputframe"_a, "inscribed"_a);
     cls.def("getNpar", &Gtransfo::getNpar);
     cls.def("offsetParams", &Gtransfo::offsetParams);
+    cls.def("computeDerivative", [](Gtransfo const &self, Point const &where, const double step) {
```
Comment (PaulPrice) [inline python/lsst/jointcal/gtransfo.cc]: `double const` is LSST standard.

File: python/lsst/jointcal/gtransfo.cc [hunk 2]
```diff
@@ -49,6 +49,11 @@ void declareGtransfo(py::module &mod) {
...
+        GtransfoLin derivative;
+        self.computeDerivative(where, derivative, step);
+        return derivative;
```
Comment (PaulPrice) [inline python/lsst/jointcal/gtransfo.cc]: I think this pattern (create a `GtransfoLin`, pass it in and then return it) indicates that the `computeDerivative` method should have an API that returns a new derivative object.

Comment (parejkoj) [reply python/lsst/jointcal/gtransfo.cc]: The existing C++ uses of `computeDerivative` almost all reuse an existing array for the output. We could look at those and see if they are worth turning into what you suggest, but that's not for this ticket.

**Pull request merged**